### PR TITLE
SS-1113 customapp copy data logs

### PR DESCRIPTION
**Pull request by hamzaimran08**
## Description

This change relates to issue [SS-1113](https://scilifelab.atlassian.net/browse/SS-1113). The logs will now be able to show copy data error logs as well. Visual adjustments have also been made to make the tabs more obvious. To test, create e.g a custom app with volume attached (to test difference use a path that exists in the container and then use a path that does not exist. The latter will cause a copy data error).

## Types of changes

To indicate the type of change (such as bugfix or new feature), use a github pull request label.

## Checklist

_If you're unsure about any of the items below, don't hesitate to ask. We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [x] I have included a link to the issue on GitHub or JIRA (if any)
- [ ] I have included migration files (if there are changes to the model classes)
- [ ] I have included, reviewed and executed tests (unit and end2end) to complement my changes
- [ ] I have updated the related documentation (if necessary)
- [ ] I have added a reviewer for this pull request
- [x] I have added myself as an author for this pull request
- [ ] In the case I have modified settings.py, then I have also updated the studio-settings-configmap.yaml file in serve-charts

## Further comments

Anything else you think we should know before merging your code!

### CODE REVIEW

File: templates/apps/logs.html [hunk 1]
```diff
@@ -8,24 +8,24 @@
...
       </div>
     <div>
       <nav>
         <div class="nav nav-tabs" id="nav-tab" role="tablist">
-          <button class="nav-link active" id="nav-contianer1-tab" data-bs-toggle="tab" data-bs-target="#nav-contianer1" type="button" role="tab" aria-controls="nav-contianer1" aria-selected="true" onclick="updateContainer('{{ instance.subdomain.subdomain }}')">{{instance.name}}</button>
+          <button class="shadow nav-link nav-tab-logs active" id="nav-contianer1-tab" data-bs-toggle="tab" data-bs-target="#nav-contianer1" type="button" role="tab" aria-controls="nav-contianer1" aria-selected="true" onclick="updateContainer('{{ instance.subdomain.subdomain }}')">{{instance.name}}</button>
```
Comment (alfredeen) [inline templates/apps/logs.html:19]: Spelling error of "container".

**Review by alfredeen — APPROVED** [review]
Spelling error, otherwise looks ok.  I was unable to test the actual logs locally but the copy data tab is displayed.

File: apps/views.py [hunk 2]
```diff
@@ -72,8 +72,8 @@ def post(self, request, project, app_slug, app_id):
         project = self.get_project(project, post=True)
         instance = self.get_instance(app_slug, app_id, post=True)
 
-        # container name is often same as subdomain name
-        container = instance.subdomain.subdomain
+        # get container name from UI (subdomain or copy-to-pvc) if none exists then use subdomain name
+        container = request.POST.get("container", "") or instance.subdomain.subdomain
```
Comment (alfredeen) [inline apps/views.py:76]: Request post data should be validated.

**Review by alfredeen — CHANGES_REQUESTED** [review]
Validate POST data before using it.

File: apps/views.py [hunk 3]
```diff
@@ -72,8 +72,13 @@ def post(self, request, project, app_slug, app_id):
...
+
+        # Perform data validation
+        if not container.isalnum() and container != "":
```
Comment (alfredeen) [inline apps/views.py:79]: This will not work for containers with names containing dashes such as "abc-def"

**Review by alfredeen — CHANGES_REQUESTED** [review]

File: apps/views.py (hunk 3 above)
Comment (alfredeen) [reply apps/views.py:79]: It could probably use SubdomainCandidateName.validate_subdomain() for consistency.

**Review by alfredeen — APPROVED** [review]
LGTM now, good work.

**Review by churnikov — APPROVED** [review]

**Pull request merged**